app/Http/Controllers/WatchlistController.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\User;
+use App\Models\Movie;
+use App\Models\MovieUser;
+use Illuminate\Http\Request;
+
+class WatchlistController extends Controller
+{
+    public function index(User $user)
+    {
+        if(auth()->user()->id == $user->id) {
```

Comment:
Same the comments above

Reply:
Or you can do if (auth()->user()->id !== $user->id) { return redirect } and the rest of the code... so you do not need the else actually, because you will return before the rest of the code is run.

Reply:
This can be done to all the code you are checking that the user is the same as the request.